packages/java/engine-core/src/main/java/dev/hilla/engine/commandrunner/GradleRunner.java

```diff
@@ -69,7 +71,14 @@ public File currentDirectory() {
 
     @Override
     public List<String> executables() {
-        return IS_WINDOWS ? List.of(".\\gradlew.bat", "gradle.bat", "gradle")
-                : List.of("./gradlew", "gradle");
+        List<String> executableList = new ArrayList<>();
+        String customExecutable = System.getProperty(EXECUTABLE_PROPERTY);
+        if (customExecutable != null) {
+            executableList.add(customExecutable);
+        }
+        executableList.addAll(
+                IS_WINDOWS ? List.of(".\\gradlew.bat", "gradle.bat", "gradle")
+                        : List.of("./gradlew", "gradle"));
```

Comment:
```suggestion
        if (IS_WINDOWS) {
            executableList.add(".\\gradlew.bat");
            executableList.add("gradle.bat");
            executableList.add("gradle");
        } else {
            executableList.add("./gradlew");
            executableList.add("gradle");
        }
```
Just a personal preference, I'd avoid creating lists just to add them.